File: server/schemas/inventory.py
from typing import Optional, List
from pydantic import BaseModel, Field

class ItemDefinitionSchema(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    type: str
    rarity: str
    icon: str
    sellValue: int
    attack: int
    defense: int
    strength: int
    knowledge: int
    discipline: int
    focus: int
    endurance: int
    recovery: int
    passive: Optional[str] = None

    class Config:
        from_attributes = True

class PlayerItemSchema(BaseModel):
    id: str
    characterId: str
    itemDefinitionId: str
    quantity: int
    isEquipped: bool
    isLocked: bool
    isFavorite: bool
    acquiredFrom: Optional[str] = None
    
    itemDefinition: Optional[ItemDefinitionSchema] = None

    class Config:
        from_attributes = True

class EquipmentActionResponse(BaseModel):
    status: str
    message: str
    playerItem: PlayerItemSchema

class ToggleActionResponse(BaseModel):
    status: str
    message: str
    playerItem: PlayerItemSchema

class ItemUseResponse(BaseModel):
    status: str
    message: str
    effectType: str
    effectValue: int
    consumedItemId: str
    remainingQuantity: int

File: client/src/features/inventory/types/inventory.ts
export type ItemType =
  | "WEAPON"
  | "HELMET"
  | "ARMOR"
  | "GLOVES"
  | "BOOTS"
  | "RING"
  | "NECKLACE"
  | "ARTIFACT"
  | "RELIC"
  | "CONSUMABLE"
  | "MATERIAL"
  | "QUEST_ITEM";

export type ItemRarity = "COMMON" | "RARE" | "EPIC" | "LEGENDARY" | "MYTHIC";

export interface ItemDefinition {
  id: string;
  name: string;
  description: string | null;
  type: ItemType;
  rarity: ItemRarity;
  icon: string;
  sellValue: number;
  attack: number;
  defense: number;
  strength: number;
  knowledge: number;
  discipline: number;
  focus: number;
  endurance: number;
  recovery: number;
  passive: string | null;
  lore?: string | null;
}

export interface PlayerItem {
  id: string;
  characterId: string;
  itemDefinitionId: string;
  quantity: number;
  isEquipped: boolean;
  isLocked: boolean;
  isFavorite: boolean;
  acquiredFrom: string | null;
  itemDefinition: ItemDefinition;
}
